lib/ecto/changeset.ex

```diff
@@ -1863,7 +1864,7 @@ defmodule Ecto.Changeset do
           query
         end
 
-      if repo.one(query) do
+      if repo.one(query, repo_opts) do
```

Comment:
Out of curiosity, what's your use case, which options do you need to pass?

Instead of adding another argument, maybe we should use the existing `opts` argument like this:

```elixir
repo_opts = Keyword.take(opts, [:prefix, :timeout, :log, :telemetry_event, :telemetry_prefix])`

if repo.one(query, opts) do
```

this way we could also remove lines 1859-1864. It would be easier to document too, we could just say see "Shared options" in Repo, same as we do in https://hexdocs.pm/ecto/Ecto.Repo.html#c:one/2-options

Reply:
See how we get the tenant ID in `prepare_query` through a 'custom' option passed to Repo in this howto:
https://github.com/elixir-ecto/ecto/blob/5904da3073832b533e87ed12a75bfbccfe7aeb66/guides/howtos/Multi%20tenancy%20with%20foreign%20keys.md

https://github.com/elixir-ecto/ecto/pull/3491#issuecomment-731763071